chromium_src/chrome/browser/ui/views/permission_bubble/permission_prompt_bubble_view.cc

```diff
@@ -87,10 +95,88 @@ void AddAdditionalWidevineViewControlsIfNeeded(
     views::BubbleDialogDelegateView* dialog_delegate_view,
     const std::vector<permissions::PermissionRequest*>& requests) {}
 #endif
+
+// Custom combobox, shows permission lifetime options and applies selected value
+// to all permissions currently visible in the bubble.
+class PermissionLifetimeCombobox : public views::Combobox,
+                                   public ui::ComboboxModel {
+ public:
+  explicit PermissionLifetimeCombobox(
+      permissions::PermissionPrompt::Delegate* delegate)
+      : delegate_(delegate),
+        lifetime_options_(permissions::CreatePermissionLifetimeOptions()) {
+    DCHECK(delegate_);
+    SetCallback(base::BindRepeating(&PermissionLifetimeCombobox::OnItemSelected,
+                                    base::Unretained(this)));
+    SetModel(this);
+    OnItemSelected();
+  }
+
+  PermissionLifetimeCombobox(const PermissionLifetimeCombobox&) = delete;
+  PermissionLifetimeCombobox& operator=(const PermissionLifetimeCombobox&) =
+      delete;
+
+  // ui::ComboboxModel:
+  int GetItemCount() const override { return lifetime_options_.size(); }
+
+  base::string16 GetItemAt(int index) const override {
+    return lifetime_options_[index].label;
+  }
+
+ private:
+  void OnItemSelected() {
+    const auto& lifetime = lifetime_options_[GetSelectedIndex()].lifetime;
+    DLOG(INFO) << "Set permission lifetime "
+               << (lifetime ? lifetime->InSeconds() : -1);
+    // TODO(https://github.com/brave/brave-browser/issues/14126): Set the
+    // lifetime for all current requests.
+    // for (auto* request : delegate_->Requests()) {
+    //   request->SetLifetime(lifetime);
+    // }
+  }
+
+  permissions::PermissionPrompt::Delegate* const delegate_;
+  std::vector<permissions::PermissionLifetimeOption> lifetime_options_;
+};
+
+void AddPermissionLifetimeComboboxIfNeeded(
+    views::BubbleDialogDelegateView* dialog_delegate_view,
+    permissions::PermissionPrompt::Delegate* delegate) {
+  if (!base::FeatureList::IsEnabled(
+          permissions::features::kPermissionLifetime)) {
+    return;
+  }
+
+  // Create a single line container for a label and a combobox.
+  auto container = std::make_unique<views::View>();
```

Comment:
nit: How about using `T* AddChildView(std::unique_ptr<T> view)`?
It's preferred over `T* AddChildView(T* view)`.

Reply:
Yep, agree, will do that.